### PR TITLE
add yaqd_core recipe

### PR DESCRIPTION
**Pull request by untzag**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by untzag** [comment]
I just noticed we don't actually distribute LICENSE...

Gonna fix this upstream, marking this as WIP for now. So sorry!

**Comment by untzag** [comment]
VERSION is distributed now

**Comment by untzag** [comment]
@ksunden could you comment confirming that you are willing to be maintainer?

**Comment by untzag** [comment]
@conda-forge/help-python

could you give me a clue about why we are failing on windows?

**Comment by npavlovikj** [comment]
@untzag , `noarch` recipes are built only on Linux, so you can ignore the Windows failures.

File: recipes/yaqd-core/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+
+requirements:
+  host:
+    - python >=3.7
+    - pip
+    - pytest-runner
```
Comment (npavlovikj) [inline recipes/yaqd-core/meta.yaml]: Usually you only need `python` and `pip` in the `host` section.

**Comment by ksunden** [comment]
I am willing to be a maintainer

File: recipes/yaqd-core/meta.yaml (hunk 1 above)
Comment (marcelotrevisani) [reply recipes/yaqd-core/meta.yaml]: It seems they do need ``pytest-runner`` in ``host`` :/
https://gitlab.com/yaq/yaqd-core-python/-/blob/master/setup.py#L24

File: recipes/yaqd-core/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
```
Comment (marcelotrevisani) [inline recipes/yaqd-core/meta.yaml]: But you need the ``entry_points`` here
```suggestion
  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
  entry_points:
    - yaqd-base=yaqd_core._daemon:Base.main
    - yaqd-hardware=yaqd_core._hardware._base:Hardware.main
    - yaqd-continuous-hardware=yaqd_core._hardware._continuous:ContinuousHardware.main
```

Thanks to [grayskull](https://github.com/marcelotrevisani/grayskull)

File: recipes/yaqd-core/meta.yaml (hunk 1 above)
Comment (marcelotrevisani) [reply recipes/yaqd-core/meta.yaml]: But you are completely right, usually just ``pip`` and ``python`` are enough

**Comment by untzag** [comment]
Thank you so much @npavlovikj @Marcello-Sega you are awesome :heart:

@conda-forge/help-python I'm sorry I let this PR sit for a month... I believe I believe I'm ready to merge but if not would appreciate any comment.

**Comment by untzag** [comment]
@conda-forge/help-python I would really appreciate any feedback

File: recipes/yaqd-core/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+test:
+  imports:
+    - yaqd_core
+
+about:
+  home: https://yaq.fyi
+  license: LGPL-3.0-only
+  license_family: LGPL
+  license_file: LICENSE
+  summary: Core python package for implementing yaq deamons, and associated utilities.
```
Comment (synapticarbors) [inline recipes/yaqd-core/meta.yaml:41]: `dev_url: https://gitlab.com/yaq/yaqd-core-python`

File: recipes/yaqd-core/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  host:
+    - python >=3.7
+    - pip
+    - pytest-runner
+  run:
+    - python >=3.7
+    - appdirs
+    - toml
+
+test:
```
Comment (synapticarbors) [inline recipes/yaqd-core/meta.yaml:32]: I think we should probably be testing the console scripts as well if feasible:

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#test-commands

Comment (untzag) [reply recipes/yaqd-core/meta.yaml:32]: The entry points start daemons that expect configuration files, so it's not super easy to test them...

Comment (untzag) [reply recipes/yaqd-core/meta.yaml:32]: I could try to run our pytest directory---going to play with that now

Comment (synapticarbors) [reply recipes/yaqd-core/meta.yaml:32]: I don't want to hold up the recipe. If it's simple to just run something like `yaqd-base -h` for each of the console scripts without setting up a bunch of config files, that would be fine. We're mostly just testing for existence. If it's going to be a lot of work, let's just skip it. No need to run the full test suite.

Comment (untzag) [reply recipes/yaqd-core/meta.yaml:32]: OK @synapticarbors I believe that's done :smile: thanks for your kind direction

**Pull request merged**